lib/linguist/heuristics.rb

```diff
@@ -127,6 +127,14 @@ def call(data)
         Language["ECL"]
       end
     end
+    
+    disambiguate ".es" do |data|
+      if /^\s*(?:%%|main\s*\(.*?\)\s*->)/.match(data)
+        Language["Erlang"]
+      elsif /(?:\/\/|("|')use strict\1|export\s+default\s|\/\*.*?\*\/)/m.match(data)
```

Comment:
@pchaigno The second expression will (the one you've quoted). I've enabled the `/m` flag so it *will* span multiple lines.

As for `var`, I'm afraid not. Authors using `es` extensions will undoubtedly be using ES6 alternatives to `var`: `let` and `const`. So that's probably less reliable; conversely, most ES6 files are written in "use strict" mode.